### PR TITLE
feat(cond): connect air conditioning to electrical system

### PR DESCRIPTION
**Pull request by mjuhe**
## Summary of Changes

This PR connects the Air Conditioning System Controller to the electrical system. It also adds a secondary channel with reduced functionality for when the main channel is unpowered (or faulted, although failures are not implemented yet). 

The ACSC is connected to:
- Primary computer: AC bus 1 + DC bus 1
- Secondary computer: AC bus 2 + DC bus 2

If primary computer is not available, the secondary computer takes over to give a reduced level of control:
- The zones are controlled to 24 degrees, the temperature can't be changed.
- The flow setting function is not available (flow is controlled to 100% / NORM)

If both computers are not available:
- The zones are fed directly by the packs, duct exhaust temperature is 20 degrees for the cockpit and 10 degrees for the cabin. No temperature control available.
- The pack flow valve controller is unpowered so all control of the valve is lost.

## Screenshots (if necessary)

image

## References

Mainly FCOM and AMM for this.

## Additional context

Again I decided to keep this PR nice and short. I implemented the responses to loss of power to the best of my understanding, but if I got something wrong please do let me know. Still to do:
- Add failures and ARINC communication
- Integrate with pressurization system
- Add trim air system

In addition, this PR doesn't add any ECAM warnings, Status memos or COND page changes (ALTN MODE / PACK REG). This will be added on a future PR (probably the one that adds the failures).

Discord username (if different from GitHub): Commander Ga#1548

## Testing instructions

A bit complicated to follow, but should be very quick to test:
1. Spawn on the runway with engines running and APU off (or gate and turn on the engines manually, but make sure no APU or EXT power is connected)
2. Activate TR1 and ESS TR failures in the electrical section of the EFB
3. On the overhead panel, move temperature dials away from middle position, validate duct temperature changes as expected
4. On the overhead panel, switch off BUS TIE and GEN1
5. In Model Behaviours validate A32NX_ELEC_AC_1_BUS_IS_POWERED is false (0) and A32NX_ELEC_AC_2_BUS_IS_POWERED is true (1)
6. This will put the ACSC in secondary channel
7. In the COND ECAM page, validate the duct temperatures changed and now it targets a cabin temp of 24 deg (you'll see this by noticing the duct temperature of a cabin zone that was below 24, now is above, and viceversa). In other words, the temperature you select on the overhead is ignored by the system as it targets 24 degrees.
8. In the BLEED page validate the pack flow valves remain open and that the overhead knob has no effect on the flow
9. On the overhead panel, switch off GEN2
10. This will unpower the system
11. Now ECAM is off, but in Model Behaviour validate duct demand is 20 degrees for cockpit and 10 deg for cabin zones (filter by COND_{ CKPT / FWD / AFT }_DUCT_TEMP)
12. Validate the packs are still supplying air (COND_PACK_FLOW_VALVE_#_IS_OPEN is true)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/systems/systems/src/air_conditioning/acs_controller.rs [hunk 1]
```diff
@@ -427,14 +476,32 @@ impl<const ZONES: usize> ZoneController<ZONES> {
...
         pack_flow: MassRate,
         pressurization: &impl Cabin,
+        operation_mode: &ACSCActiveComputer,
     ) {
-        self.zone_selected_temperature = acs_overhead.selected_cabin_temperature(self.zone_id);
+        self.zone_selected_temperature = if matches!(operation_mode, ACSCActiveComputer::Secondary)
+        {
+            // If the Zone controller is working on secondary power, the zones are controlled to
+            // 24 degrees by the secondary computer
+            ThermodynamicTemperature::new::<degree_celsius>(24.)
```
Comment (crocket63) [inline src/systems/systems/src/air_conditioning/acs_controller.rs:485]: Not sure how this module could be ported to A380, but having those temperatures as construction parameter could help using this module for either plane (if temperatures are different). If it's totally irrelevant for A380 forget what I just said :)

**Review by crocket63 — APPROVED** [review]
Sounds good!

File: src/systems/systems/src/air_conditioning/acs_controller.rs (hunk 1 above)
Comment (mjuhe) [reply src/systems/systems/src/air_conditioning/acs_controller.rs:485]: Very good point. I'm just looking into the A380 air con now (which I probably should've done long ago to be honest) and there will be some differences that we'll need to take into account. I will need to go through it in a lot more detail to understand what's common and what isn't though, I suggest leaving it for a future PR :)

**Comment by 2hwk** [comment]
QA Report

Discord: 2Cas#1022
Object of testing: #7415
Tier of Testing: 1
Date: 28/08/2022

- [x]  Spawn on the runway with engines running and APU off (or gate and turn on the engines manually, but make sure no APU or EXT power is connected)
- [x]  Activate TR1 and ESS TR failures in the electrical section of the EFB
- [x]  On the overhead panel, move temperature dials away from middle position, validate duct temperature changes as expected
- [x]  On the overhead panel, switch off BUS TIE and GEN1
- [x]  In Model Behaviours validate A32NX_ELEC_AC_1_BUS_IS_POWERED is false (0) and A32NX_ELEC_AC_2_BUS_IS_POWERED is true (1)
- [x]  This will put the ACSC in secondary channel
- [x]  In the COND ECAM page, validate the duct temperatures changed and now it targets a cabin temp of 24 deg (you'll see this by noticing the duct temperature of a cabin zone that was below 24, now is above, and viceversa). In other words, the temperature you select on the overhead is ignored by the system as it targets 24 degrees.
- [x]  In the BLEED page validate the pack flow valves remain open and that the overhead knob has no effect on the flow
- [x]  On the overhead panel, switch off GEN2
- [x]  This will unpower the system
- [x]  Now ECAM is off, but in Model Behaviour validate duct demand is 20 degrees for cockpit and 10 deg for cabin zones (filter by COND_{ CKPT / FWD / AFT }_DUCT_TEMP)
- [x]  Validate the packs are still supplying air (COND_PACK_FLOW_VALVE_#_IS_OPEN is true)


Negatives: N/A

Testing Results: Passed

Conclusions: LGTM

**Pull request merged**